coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -594,6 +594,15 @@ func (s *server) acquireProtoJob(ctx context.Context, job database.ProvisionerJo
 			})
 		}
 
+		roles, err := s.Database.GetAuthorizationUserRoles(dbauthz.AsSystemRestricted(ctx), owner.ID)
```

Comment:
The `ctx` should already have the right perms. If it does not, we can add what it needs to here: https://github.com/coder/coder/blob/main/coderd/database/dbauthz/dbauthz.go#L164-L193
```suggestion
		roles, err := s.Database.GetAuthorizationUserRoles(ctx, owner.ID)
```